Migrate Block Mover Test For Playwright

pooja-muchandikar opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
## What?
<!-- In a few words, what is the PR actually doing? -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ block-mover.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/block-mover.test.js) to its Playwright version.
## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
Run` npm run test-e2e:playwright test/e2e/specs/editor/various/block-mover.spec.js`

pooja-muchandikar committed:
Migrate Block Mover Test For Playwright

pooja-muchandikar committed:
Update styling issue

pooja-muchandikar committed:
Resolve errors

pooja-muchandikar commented:
Hi @kevin940726,

I hope you are doing well!

Could you please help me in reviewing this PR?

Thanks!!

talldan commented on test/e2e/specs/editor/various/block-mover.spec.js:
```diff
@@ -0,0 +1,50 @@
+		page,
+	} ) => {
+		// Create a two blocks on the page.
+		await page.click( '.block-editor-default-block-appender' );
+		await page.keyboard.type( 'First Paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second Paragraph' );
```
I think it would be good to simplify this part with something like:
```js
await editor.insertBlock( { name: 'core/paragraph', attributes: { content: 'First Paragraph' } } );
await editor.insertBlock( { name: 'core/paragraph', attributes: { content: 'Second Paragraph' } } );

pooja-muchandikar replied:
Worked on the feedbacks

talldan commented on test/e2e/specs/editor/various/block-mover.spec.js:
```diff
@@ -0,0 +1,50 @@
+		// Select a block so the block mover is rendered.
+		await page.focus( 'text=First Paragraph' );
+		await editor.showBlockToolbar();
+		const count = await page.$$eval(
+			'.block-editor-block-mover',
+			( el ) => el.length
+		);
+		expect( count ).toBe( 1 );
```
It'd be good to take advantage of Playwright's locator system here, and use the role selector:
```js
const moveDownButton = page.locator( 'role=toolbar[name="Block tools"i] >> role=button[name="Move down"i]' );
const moveUpButton = page.locator( 'role=toolbar[name="Block tools"i] >> role=button[name="Move up"i]' );
await expect( moveDownButton ).toBeVisible();
await expect( moveUpButton ).toBeVisible();
```

pooja-muchandikar replied:
Worked on the feedbacks

talldan commented on test/e2e/specs/editor/various/block-mover.spec.js:
```diff
@@ -0,0 +1,50 @@
+		page,
+	} ) => {
+		// Create a single block on the page.
+		await page.click( '.block-editor-default-block-appender' );
+		await page.keyboard.type( 'First Paragraph' );
```
Similar to the previous comment, this can change to:
```js
await editor.insertBlock( { name: 'core/paragraph', attributes: { content: 'First Paragraph' } );
```

pooja-muchandikar replied:
Worked on the feedbacks

talldan commented on test/e2e/specs/editor/various/block-mover.spec.js:
```diff
@@ -0,0 +1,50 @@
+		await editor.showBlockToolbar();
+
+		// Ensure no block mover exists when only one block exists on the page.
+		const count = await page.$$eval(
+			'.block-editor-block-mover',
+			( el ) => el.length
+		);
+		expect( count ).toBe( 0 );
```
And this part could also be refactored to be use locators:
```js
const moveDownButton = page.locator( 'role=toolbar[name="Block tools"i] >> role=button[name="Move down"i]' );
const moveUpButton = page.locator( 'role=toolbar[name="Block tools"i] >> role=button[name="Move up"i]' );
await expect( moveDownButton ).toBeHidden();
await expect( moveUpButton ).toBeHidden();
```

pooja-muchandikar replied:
Worked on the feedbacks.

talldan reviewed (commented):
Thanks for working on this! I left a few comments about how we can take advantage playwright's nicer API. I haven't tested the code snippets, so you might need to double check they're correct.

pooja-muchandikar committed:
Addressed Review Feedback

pooja-muchandikar commented:
Hi @talldan,

I have addressed all the feedbacks, could you please review again?

Also I noticed that CI is failing but not related to the test case I worked on. I am little confused here 🤔

![image](https://user-images.githubusercontent.com/41000648/176881962-039c9403-e040-4c7b-a667-963a978e8bc5.png)

---

![image](https://user-images.githubusercontent.com/41000648/176881982-a7e2b14f-b8af-4d7b-8656-ef2d5944dae4.png)

talldan commented:
Thanks for making those changes!

> Also I noticed that CI is failing but not related to the test case I worked on. I am little confused here

Some of our old tests are not very reliable, which is one of the reasons for migrating them to Playwright.

I've restarted them to see if they pass.

pooja-muchandikar commented:
Hi @talldan, good morning..

All the CI checks have passed except the End to End Tests / Admin -3 🤔 

![image](https://user-images.githubusercontent.com/41000648/177244317-28cd9a5b-396b-41a4-9eee-92eebab58574.png)

talldan commented:
Did a another retry and managed to get it passing 👍

talldan reviewed (approved):
Thanks for working on this!

pooja-muchandikar commented:
Thank you :)
